src/main/java/io/cdap/plugin/kinesis/streaming/ReferenceStreamingSource.java

```diff
@@ -58,4 +66,28 @@ public void run(DatasetContext datasetContext) {
     });
   }
 
+  /**
+   * Record field-level lineage for streaming source plugins (ReadOperation). This method should be called from
+   * prepareRun of any streaming source plugin.
+   *
+   * @param context StreamingSourceContext from prepareRun
+   * @param outputName name of output dataset
+   * @param tableSchema schema of fields. Also used to determine list of field names. Schema and schema.getFields() must
+   * not be null.
+   * @param operationName name of the operation
+   * @param description operation description; complete sentences preferred
+   */
+  protected void recordLineage(StreamingSourceContext context, String outputName, Schema tableSchema,
+                               String operationName, String description)
+    throws DatasetManagementException, TransactionFailureException {
+    Preconditions.checkNotNull(tableSchema, "schema for output %s is null.", outputName);
```

Comment:
Don't use Precondition, if will fail the pipeline. Should just do a if check and log as a warn message that field lineage will be missing